plugin/pkg/shoot/dnshostedzone/admission.go

```diff
@@ -142,8 +142,14 @@ func (d *DNSHostedZone) Admit(a admission.Attributes) error {
 		return nil
 	}
 
+	// Checks on Hosted Zone ID can only be done if the Shoot manifest has a domain.
+	if shoot.Spec.DNS.Domain == nil {
+		dnsPath := field.NewPath("spec", "dns", "domain")
+		return field.Required(dnsPath, "domain field is required")
```

Comment:
why not `apierrors.NewBadRequest("shoot domain field .spec.dns.domain must be set if provider!=unmanaged")`?

Reply:
Or 
```
admission.NewForbidden(a, field.Required(dnsPath, "domain field is required"))
```
to adjust to `plugin/pkg/shoot/validator/admission.go`?

Reply:
Forbidden would not be correct, right? It's a bad request, the user sent an invalid manifest. I'm fine with `admission.BadRequest(field.Required(...))` if that's possible.

Reply:
Sure!

> shoot domain field .spec.dns.domain must be set if provider!=unmanaged

I'd prefer to leave out the part with `if provider!=unmanaged` because from a user perspective, only setting it to `unmanaged` would result in an error of the next admission plugin. Would you agree?

Reply:
Which error do you mean?

Reply:
Never mind, I missed the defaulting logic :)